src/state/CallViewModel/LayoutSwitch.ts

```diff
@@ -59,7 +59,13 @@ export function createLayoutModeSwitch(
         // To allow the user to override the auto-switch by selecting grid mode again.
         scan<
           [GridMode, boolean, WindowMode],
-          { mode: GridMode; hasAutoSwitched: boolean }
+          {
+            mode: GridMode;
+            /** Remember if the change was user driven or not */
+            hasAutoSwitched: boolean;
+            /** To know if it is new screen share or an already handled */
+            prevShare: boolean;
+          }
         >(
           (acc, [userSelection, hasScreenShares, windowMode]) => {
```

Comment:
I would find it slightly more readable to call prevShare "hasScreenShares" (since this is exactly what it reflects) and acc "prev" (since this stateful variable is where the idea of these values being "previous" comes in)

```suggestion
            hasScreenShares: boolean;
          }
        >(
          (prev, [userSelection, hasScreenShares, windowMode]) => {
```